cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleSpecification.java

```diff
@@ -27,16 +27,22 @@
  */
 public final class ScheduleSpecification {
   private final Schedule schedule;
+  private final String scheduleType;
```

Comment:
Why would you need that extra argument? If you actually look at how ScheduleSpecification is serialized in JSON from CDAP - in ``ScheduleSpecificationCodec`` - a scheduleType param is already added to the json. You could actually reuse this SerDe, instead of creating your own ``ScheduleSpecificationDeserializer``.
Also if you add an attribute here, isn't it going to mess with existing app specifications present in CDAP?